docs/topics/Automated-Checks.md

```diff
@@ -0,0 +1,259 @@
+
+1. `NeedsApproval`: The tournament is submitted but waiting approval from a verifier.
+2. `NeedsMatchData`: Match data needs to be fetched via the osu! API.
+3. `NeedsAutomationChecks`: The tournament, and all of its children, are awaiting automation checks.
+4. `NeedsVerification`: Awaiting human review
+5. `NeedsStatCalculation`: After human review, process statistics (must be complete before it is eligible for inclusion in the rating system).
+6. `Done`: Processing is completed. `Verified` tournaments with this status are eligible for inclusion in the rating system.
+
+### `RejectionReason`
+
+Each entity has a custom `RejectionReason` type with various flags which may cause it to be marked as `PreRejected`. Flags can be combined with each other to form a set of reasons. For example, a `Game` could be marked as `PreRejected` by the system due to `NoScores` and `BeatmapNotPooled`.
+
+## Flow
+
+Automation checks are performed in the following order.
+
+```Mermaid
+flowchart LR;
+    GameScore --> Game --> Match --> Tournament
+```
+
+> This allows the parent entities to have the context of how their children faired during the automated checks process.
+> 
+
+### Tournament
+
+```Mermaid
+flowchart TD;
+   A[Is the count of PreVerified and/or Verified matches >= 0?]
+   B[Apply NoVerifiedMatches flag to RejectionReason]
+   C[Is this count >= 80% of the total match count?]
+   D[Apply NotEnoughVerifiedMatches flag to RejectionReason]
+   PreTerm[Is the RejectionReason null?]
+   TermPositive[Change VerificationStatus to PreVerified]
+   TermNegative[Change VerificationStatus to PreRejected]
+   
+   A -- No --> B --> PreTerm
+   A -- Yes --> C
+   C -- No --> D --> PreTerm
+   C -- Yes --> PreTerm
+   PreTerm -- Yes --> TermPositive
+   PreTerm -- No --> TermNegative
+```
+
+### Match
+
+```Mermaid
+flowchart TD;
+   A[Is the count of games > 2?]
+   B[Do any games besides the first 2 have a 
+      RejectionReason of BeatmapNotPooled?]
+   C[Apply UnexpectedBeatmapsFound to WarningFlags]
+   D[Is the EndTime property equal to 
+   2007-09-17-00:00:00?]
+   E[Apply NoEndTime flag to RejectionReason]
+   H[Is the match name structured in a typical 
+      format?]
+   I[Apply UnexpectedNameFormat to WarningFlags]
+   J[Does the match name start with the tournament's 
+      abbreviation?]
+   K[Apply NamePrefixMismatch flag to RejectionReason]
+   L[Is the tournament's lobby size equal to 1?]
+   M[Are the games structured in a way which supports 
+      conversion to TeamVS?]
+   N[Attempt to convert a full set of Head to Head games to TeamVS]
+   O[Apply FailedTeamVsConversion flag to RejectionReason, repeat 
+      for all child games]
+   P[Convert all games to TeamVS, mark all games as PreVerified]
+   
+   F[Is the count of games equal to 0?]
+   G[Apply NoGames flag to RejectionReason]
+   Q[What is the count of PreVerified and/or Verified games?]
+   Q1[0]
+   Q2[1 or 2]
+   Q3[4 or 5]
```

Comment:
check if 5 is included in this case or the other one
```suggestion
   Q3[3 or 4]
```

Reply:
Q2 should be `Q2[1 to 3]` instead.